receiver/otelarrowreceiver/internal/sharedcomponent/doc.go

```diff
@@ -0,0 +1,8 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Package sharedcomponent exposes util functionality for receivers and exporters
+// that need to share state between different signal types instances such as net.Listener or os.File.
+//
+// This package was copied from collector/internal/sharedcomponent.
+package sharedcomponent // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/otelarrowreceiver/internal/sharedcomponent"
```

Comment:
We have https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/internal/sharedcomponent as well, is there a reason to reproduce it here?

Reply:
No reason. I didn't know about that module, and I see it has a different interface than the one in core. I wonder, of course, why it was reproduced here instead of referring to the core module, and/or whether that should change eventually. The implementation in core allows the component "new" function to return err, and this library doesn't, so this will be a surprisingly non-trivial change.

Reply:
Done. However, returning a bug report: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30924